ecosystem/typescript/sdk/src/bcs/helper.ts

```diff
@@ -45,6 +45,32 @@ export function deserializeVector(deserializer: Deserializer, cls: any): any[] {
   return list;
 }
 
+// Recursive function to serialize n-dimensional arrays
+function serializeNestedArrayWithFunc(value: any[], func: string, serializer: Serializer): void {
+  if (Array.isArray(value[0])) {
+    serializer.serializeU32AsUleb128(value.length)
```

Comment:
`serializer.serializeU32AsUleb128(value.length)` is in both the `if` and the `else` branches. That means we can move it out.

Reply:
I accidentally lost the previous comment. It seems that rebase is not optional.